### ISSUE TITLE
Language not recognized properly

### ISSUE THREAD

**Issue by energyuser** [validator/validator#491]
So I validated my site and saw a strange warning:
>This document appears to be written in Russian but the value of the HTTP Content-Language header is bg. Consider changing it to ru (or variant).

But the site IS in BG [Bulgarian], not Russian. Why doesn't the validator respect the `lang` attribute of the `<html>` element? What makes it think that's actually Russian? And last but not least, since when is it the validator's job to check if `lang` matches the page language?

[Link to my validated site (the warning may not appear every time, try F5 if you can't see it)](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=http%3A%2F%2Fwww.myradiorestorations.info%2Fcategory-1%2F)

**Comment by sideshowbarker** [comment]
I pushed a fix for this to the HTML checker (validator), so I think you won’t get that error now.

>  And last but not least, since when is it the validator's job to check if `lang` matches the page language?

I added the language-detection feature in July 2016. See https://github.com/validator/validator/commit/d1ee5cdb941051625e29ce4ac6ca44b2f2a4b38a:

> This runs https://github.com/shuyo/language-detection on every document
checked, and if the document has an `html[lang]` value that does not match
the detected language, a new warning is emitted; and a related warning is
emitted if the document has no `html[lang]` at all.
> 
> The goal is to help authors avoid the mistake of, for example, having a
document labeled `html[lang=en]` whose actual contents are actually in a
language other than English. Such lang-mislabeled documents cause
particular problems for screen-reader users; for some background, see:
>
>  https://www.w3.org/Bugs/Public/show_bug.cgi?id=26942#c5

Unfortunately the https://github.com/shuyo/language-detection library that does the actual language detection doesn’t always do so great at accurately guessing particular languages, and Bulgarian is one that it seems to have difficulties with.

So to get around the limitations in that library, the change I made today causes the HTML checker not emit any warning if a document has `html[lang=bg]` (like yours does) but the language detector (mis)guesses that it’s written in Russian.
